### PR TITLE
Add r-keyring

### PR DESCRIPTION
**Pull request by halldc**
Adding a recipe for the [keyring](https://cran.r-project.org/package=keyring) R package. This was created using the [conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper), and then I added the `libsecret` requirement for Linux.

Depends on https://github.com/conda-forge/staged-recipes/pull/6894, https://github.com/conda-forge/staged-recipes/pull/6895 and https://github.com/conda-forge/r-getpass-feedstock/pull/2

### CODE REVIEW

**Comment by halldc** [comment]
@cbrueffer @bgruening Now that staged-recipes has become unclogged, this PR looks ready to merge.

File: recipes/r-keyring/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    - {{native}}toolchain        # [win]
+    - {{posix}}filesystem        # [win]
+    - {{posix}}sed               # [win]
+    - {{posix}}grep              # [win]
+    - {{posix}}autoconf
+    - {{posix}}automake          # [not win]
+    - {{posix}}automake-wrapper  # [win]
+    - {{posix}}pkg-config
+    - {{posix}}make
+    - {{posix}}sed  # [win]
```
Comment (cbrueffer) [inline recipes/r-keyring/meta.yaml]: sed and coreutils are duplicated

Comment (cbrueffer) [reply recipes/r-keyring/meta.yaml]: I can merge directly after the fix, no need to wait for the CI

Comment (halldc) [reply recipes/r-keyring/meta.yaml]: Ah, I'm sorry I didn't notice that myself. I guess I should be on the lookout in future.

File: recipes/r-keyring/meta.yaml [hunk 2]
```diff
@@ -32,8 +32,6 @@ requirements:
     - {{posix}}automake-wrapper  # [win]
     - {{posix}}pkg-config
     - {{posix}}make
-    - {{posix}}sed  # [win]
```
Comment (cbrueffer) [inline recipes/r-keyring/meta.yaml]: Thanks! Maybe we should just fix the template :-D

**Pull request merged**